gateway/gateway.go

```diff
@@ -27,27 +29,24 @@ type remoteService struct {
 	gatewayrpc.Service
 	*url.URL
 	origURL string
-	srv     bool
 }
 
-// Request contains all the data about an incoming request which is currently
-// known. It can be used in conjunction to RequestCallback in order to provide
-// extra functionality or checks to request coming through.
-//
-// Notably, rpc.CodecRequest can be used to read in the arguments of the call
-// and write back responses or errors to the client if no forwarding should be
-// done.
-//
-// Also, the Body of the *http.Request will be filled with the raw body of
-// content which originally came in, and there's no need to actually close it.
-// the *http.Request's URL field will also have been modified to reflect the url
-// the request will be being forwarded to.
-type Request struct {
-	http.ResponseWriter
-	*http.Request
-	ServiceName string
-	gatewayrpc.Method
-	rpc.CodecRequest
+type externalHandler struct{}
```

Comment:
instead of making an empty struct you could do something like:

```
var externalHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) { ... })
```